src/NUnitFramework/tests/TestUtilities/Comparers/GenericEqualityComparison.cs

```diff
@@ -0,0 +1,43 @@
+﻿// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+
+namespace NUnit.TestUtilities.Comparers
+{
+    internal class GenericEqualityComparison<T>
+    {
+        public bool WasCalled = false;
+
+        public Func<T, T, bool> Delegate
+        {
+            get { return new Func<T, T, bool>(Compare); }
+        }
```

Comment:
Totally up to your personal style, but to point out an optional improvement in case you weren't aware:
 `new Func<T, T, bool>` can be implicit here, and since we now allow C# 6, this generates the exact same output:
`public Func<T, T, bool> Delegate => Compare;`. No need to change though.

Reply:
I agree with the last sentence. 😄 I like the explicitness of it when I'm reading the code.

Reply:
I actually hadn't thought of the C# 6 expression-bodied member, thanks for the note. I think I'll leave it as is, though, since I also find it nice and explicit this way.